### PR TITLE
Updates to generator refactoring, catchpoint and indexer support

### PR DESCRIPTION
**Pull request by algonautshant**
* Proposed changes to Will's review: generator refactoring #130
 ** Generator spec file path as optional parameter.
 ** Small fixes

* Support for Indexer catchpoint http methods.
* Updates from the new spec file.

### CODE REVIEW

**Comment by winder** [comment]
LGTM. But I think that we might be excluding the "private" endpoints from the SDKs, the node status definitely still needed to be updated though.

File: generator/src/main/java/com/algorand/sdkutils/RunQueryMapperGenerator.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ public static void main (String args[]) throws JsonProcessingException, IOExcept
 
 		
 		QueryMapperGenerator qmg = new QueryMapperGenerator(indexerRoot, algodRoot);
-		qmg.writeQueryMapper("src/test/java/com/algorand/algosdk/unit/utils");
+		qmg.writeQueryMapper("../src/test/java/com/algorand/algosdk/unit/utils");
```
Comment (winder) [inline generator/src/main/java/com/algorand/sdkutils/RunQueryMapperGenerator.java:38]: I was thinking that these would go away with the CLI mode. Left them here mainly for example and convenience.

**Comment by winder** [comment]
Are there missing files? I thought I commented out several QueryMapper components

**Comment by winder** [comment]
I'm merging this in, but we may need a followup to remove the catchup endpoint (and participation keys/shutdown endpoints from earlier)

File: generator/src/main/java/com/algorand/sdkutils/RunQueryMapperGenerator.java (hunk 1 above)
Comment (algonautshant) [reply generator/src/main/java/com/algorand/sdkutils/RunQueryMapperGenerator.java:38]: I am definitely in favor or leaving the hard-coded "default" paths.

**Pull request merged**